ext/CEGUI/IrrlichtBaW/IrrlichtBaWRenderTarget.cpp

```diff
@@ -0,0 +1,48 @@
+#include "IrrlichtBaWRenderTarget.hpp"
+
+using namespace irr;
+using namespace ext;
+using namespace cegui;
+
+const CEGUI::String IrrlichtBaWRenderTarget::EventNamespace = "?";		// TODO
+const CEGUI::String IrrlichtBaWRenderTarget::EventAreaChanged = "?";	// TODO
+
+void IrrlichtBaWRenderTarget::draw(const CEGUI::GeometryBuffer& buffer)
+{
+
+}
+
+void IrrlichtBaWRenderTarget::draw(const CEGUI::RenderQueue& queue)
+{
+
+}
+
+void IrrlichtBaWRenderTarget::setArea(const CEGUI::Rectf& area)
+{
+
+}
+
+const CEGUI::Rectf& IrrlichtBaWRenderTarget::getArea() const
+{
+
+}
+
+bool IrrlichtBaWRenderTarget::isImageryCache() const 
+{
+
+}
+
+void IrrlichtBaWRenderTarget::activate() 
+{
+
+}
+
+void IrrlichtBaWRenderTarget::deactivate() 
+{
+
+}
```

Comment:
these will probably be FBO bind and unbind, check if OpenGL backend does more than that

Reply:
it sets viewport 

```C++
template <typename T>
void OpenGLRenderTarget<T>::activate()
{
    glViewport(static_cast<GLsizei>(d_area.left()),
               static_cast<GLsizei>(d_area.top()),
               static_cast<GLsizei>(d_area.getWidth()),
               static_cast<GLsizei>(d_area.getHeight()));

    if (!d_matrixValid)
        updateMatrix();

    d_owner.setViewProjectionMatrix(d_matrix);

    d_owner.setActiveRenderTarget(this);
}

//----------------------------------------------------------------------------//
template <typename T>
void OpenGLRenderTarget<T>::deactivate()
{
}
```

Reply:
that would be a:
https://github.com/buildaworldnet/IrrlichtBAW/blob/shader_pipeline/include/IVideoDriver.h#L139
https://github.com/buildaworldnet/IrrlichtBAW/blob/shader_pipeline/include/IVideoDriver.h#L112

The matrix stuff you need to compute yourself and send to shader via Push Constant or UBO